src/NHibernate/Driver/SqlServerCeDriver.cs

```diff
@@ -104,8 +109,8 @@ protected override void InitializeParameter(DbParameter dbParam, string name, Sq
 			AdjustDbParamTypeForLargeObjects(dbParam, sqlType);
 			if (prepareSql)
 			{
-				SqlClientDriver.SetVariableLengthParameterSize(dbParam, sqlType);
-		}
+				SqlClientDriver.SetVariableLengthParameterSize(dbParam, sqlType, _dbDecimalPrecision, _dbDecimalScale);
+			}
```

Comment:
Does it really makes sens to set length for SqlServerCe? I believe its sole purpose it to optimize query plan cache usage, but does SqlServerCe have a query plan cache?